frontend/packages/kubevirt-plugin/src/components/vms/vm-page-new.tsx

```diff
@@ -127,18 +127,24 @@ const VMRow: React.FC<RowFunctionArgs<VMRowObjType, VmStatusResourcesValue>> = (
   customData: vmStatusResources,
 }) => {
   const { vm, vmi } = obj;
-  const { name, namespace, creationTimestamp, uid, node } = obj.metadata;
+  const { name, namespace, creationTimestamp, node } = obj.metadata;
   const dimensify = dimensifyRow(tableColumnClasses);
   const arePendingChanges = hasPendingChanges(vm, vmi);
   const printableStatus = obj?.metadata?.status;
   const status: VMStatus = printableToVMStatus?.[printableStatus];
 
   let model: K8sKind;
+  let objReference;
+  let context;
 
   if (vm) {
     model = VirtualMachineModel;
+    objReference = referenceFor(vm);
+    context = { [objReference]: vm };
   } else if (vmi) {
     model = VirtualMachineInstanceModel;
+    objReference = referenceFor(vmi);
+    context = { [objReference]: vmi };
   }
```

Comment:
consider:
```suggestion
  let model: K8sKind;
  let objReference;
  let context;

  if (vm) {
    model = VirtualMachineModel;
    objReference = referenceFor(vm);
    context = { [objReference]: vm };
  } else if (vmi) {
    model = VirtualMachineInstanceModel;
    objReference = referenceFor(vmi);
    context = { [objReference]: vmi };
  }
 
```

const dataBuilder = (vm: VMKind, vmi: VMIKind) => {
    return {
      model: (vm && VirtualMachineModel) || (vmi && VirtualMachineInstanceModel),
      objReference: referenceFor(vm || vmi),
      context: { [objReference]: vm || vmi },
    };
  };`

same goes to 2 other files where you use similar